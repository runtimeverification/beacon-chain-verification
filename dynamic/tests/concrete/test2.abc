stateTransition(#Block(( 1,10001),( 0,    0),,,,.VoluntaryExits))
stateTransition(#Block(( 4,10004),( 1,10001),,,,.VoluntaryExits))
stateTransition(#Block(( 5,10005),( 4,10004),,#Attestation(0,4,(0,0),(1,10004),0,0,0)
                                              #Attestation(1,4,(0,0),(1,10004),0,0,0)
                                              #Attestation(2,4,(0,0),(1,10004),0,0,0)
                                              #Attestation(3,4,(0,0),(1,10004),0,0,0)
                                              #Attestation(4,4,(0,0),(1,10004),0,0,0)
,,.VoluntaryExits))
processSlots(8)
