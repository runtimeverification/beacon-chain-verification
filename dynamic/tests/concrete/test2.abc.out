<T>
  <k>
    .
  </k>
  <currentSlot>
    8
  </currentSlot>
  <states>
    <state>
      <slot>
        0
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
      </attested>
      <justified>
        0 |-> some 0
      </justified>
      <finalized>
        0 |-> some 0
      </finalized>
      <lastBlock>
        ( 0 , 0 )
      </lastBlock>
      <lastJustified>
        ( 0 , 0 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state> <state>
      <slot>
        1
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
      </attested>
      <justified>
        0 |-> some 0
      </justified>
      <finalized>
        0 |-> some 0
      </finalized>
      <lastBlock>
        ( 1 , 10001 )
      </lastBlock>
      <lastJustified>
        ( 0 , 0 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state> <state>
      <slot>
        2
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
      </attested>
      <justified>
        0 |-> some 0
      </justified>
      <finalized>
        0 |-> some 0
      </finalized>
      <lastBlock>
        ( 1 , 10001 )
      </lastBlock>
      <lastJustified>
        ( 0 , 0 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state> <state>
      <slot>
        3
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
      </attested>
      <justified>
        0 |-> some 0
      </justified>
      <finalized>
        0 |-> some 0
      </finalized>
      <lastBlock>
        ( 1 , 10001 )
      </lastBlock>
      <lastJustified>
        ( 0 , 0 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state> <state>
      <slot>
        4
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
        1 |-> .Attestations
      </attested>
      <justified>
        0 |-> some 0
        1 |-> none
      </justified>
      <finalized>
        0 |-> some 0
        1 |-> none
      </finalized>
      <lastBlock>
        ( 4 , 10004 )
      </lastBlock>
      <lastJustified>
        ( 0 , 0 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state> <state>
      <slot>
        5
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
        1 |-> #Attestation ( 90004 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90003 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90002 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90001 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90000 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  .Attestations
      </attested>
      <justified>
        0 |-> some 0
        1 |-> none
      </justified>
      <finalized>
        0 |-> some 0
        1 |-> none
      </finalized>
      <lastBlock>
        ( 5 , 10005 )
      </lastBlock>
      <lastJustified>
        ( 0 , 0 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state> <state>
      <slot>
        6
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
        1 |-> #Attestation ( 90004 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90003 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90002 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90001 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90000 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  .Attestations
      </attested>
      <justified>
        0 |-> some 0
        1 |-> none
      </justified>
      <finalized>
        0 |-> some 0
        1 |-> none
      </finalized>
      <lastBlock>
        ( 5 , 10005 )
      </lastBlock>
      <lastJustified>
        ( 0 , 0 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state> <state>
      <slot>
        7
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
        1 |-> #Attestation ( 90004 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90003 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90002 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90001 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90000 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  .Attestations
      </attested>
      <justified>
        0 |-> some 0
        1 |-> none
      </justified>
      <finalized>
        0 |-> some 0
        1 |-> none
      </finalized>
      <lastBlock>
        ( 5 , 10005 )
      </lastBlock>
      <lastJustified>
        ( 0 , 0 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state> <state>
      <slot>
        8
      </slot>
      <validators>
        90000 |-> #Validator ( 90000 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90001 |-> #Validator ( 90001 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90002 |-> #Validator ( 90002 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90003 |-> #Validator ( 90003 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90004 |-> #Validator ( 90004 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90005 |-> #Validator ( 90005 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
        90006 |-> #Validator ( 90006 , false , ( 1 , 1 ) , ( 0 , 0 ) , ( 999999 , 999999 ) )
      </validators>
      <slashedBalance>
        0
      </slashedBalance>
      <attested>
        0 |-> .Attestations
        1 |-> #Attestation ( 90004 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90003 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90002 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90001 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90000 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  .Attestations
        2 |-> .Attestations
      </attested>
      <justified>
        0 |-> some 0
        1 |-> some 10004
        2 |-> none
      </justified>
      <finalized>
        0 |-> some 0
        1 |-> none
        2 |-> none
      </finalized>
      <lastBlock>
        ( 5 , 10005 )
      </lastBlock>
      <lastJustified>
        ( 1 , 10004 )
      </lastJustified>
      <lastFinalized>
        ( 0 , 0 )
      </lastFinalized>
    </state>
  </states>
  <blocks>
    <block>
      <bSlot>
        0
      </bSlot>
      <bID>
        0
      </bID>
      <parent>
        ( -1 , -1 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        .Attestations
      </attestations>
      <deposits>
        .Deposits
      </deposits>
      <exits>
        .VoluntaryExits
      </exits>
    </block> <block>
      <bSlot>
        1
      </bSlot>
      <bID>
        10001
      </bID>
      <parent>
        ( 0 , 0 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        .Attestations
      </attestations>
      <deposits>
        .Deposits
      </deposits>
      <exits>
        .VoluntaryExits
      </exits>
    </block> <block>
      <bSlot>
        4
      </bSlot>
      <bID>
        10004
      </bID>
      <parent>
        ( 1 , 10001 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        .Attestations
      </attestations>
      <deposits>
        .Deposits
      </deposits>
      <exits>
        .VoluntaryExits
      </exits>
    </block> <block>
      <bSlot>
        5
      </bSlot>
      <bID>
        10005
      </bID>
      <parent>
        ( 4 , 10004 )
      </parent>
      <slashings>
        .Slashings
      </slashings>
      <attestations>
        #Attestation ( 90000 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90001 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90002 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90003 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  #Attestation ( 90004 , 4 , ( 0 , 0 ) , ( 1 , 10004 ) )  .Attestations
      </attestations>
      <deposits>
        .Deposits
      </deposits>
      <exits>
        .VoluntaryExits
      </exits>
    </block>
  </blocks>
</T>
